spec/lib/export/dwca/data_spec.rb

```diff
@@ -451,42 +654,69 @@
-
           f.update!(collecting_event: c)
 
+          # Only create data for p1 and p3, not p2
           d1 = FactoryBot.create(:valid_data_attribute_internal_attribute, attribute_subject: c, predicate: p1 )
           d2 = FactoryBot.create(:valid_data_attribute_internal_attribute, attribute_subject: c, predicate: p3 )
 
+          # Configure all 3 predicates but only 2 have data
           a = Export::Dwca::Data.new(core_scope: scope, predicate_extensions: {collecting_event_predicate_id: predicate_ids } )
 
-          expect(a.used_predicates).to contain_exactly("TW:DataAttribute:CollectingEvent:#{p1.name}", "TW:DataAttribute:CollectingEvent:#{p3.name}")
+          predicate_file = a.predicate_data
+          predicate_file.rewind
+          content = predicate_file.read
+
+          # Parse the TSV header
+          headers = content.lines.first.strip.split("\t")
+
+          # Should only have columns for predicates with data (p1 and p3), not p2
+          expect(headers).to contain_exactly(
+            "TW:DataAttribute:CollectingEvent:#{p1.name}",
+            "TW:DataAttribute:CollectingEvent:#{p3.name}"
+          )
+
+          # Should NOT include p2 since it has no data
+          expect(headers).not_to include("TW:DataAttribute:CollectingEvent:#{p2.name}")
 
           a.cleanup
         end
 
-        specify '#used_predicates 2' do
+        specify 'predicate_data sanitizes newlines and tabs in values' do
           f = Specimen.first
 
           c = FactoryBot.create(:valid_collecting_event)
-          d4 = FactoryBot.create(:valid_data_attribute_internal_attribute, attribute_subject: c, predicate: p1 )
-
           f.update!(collecting_event: c)
 
-          d1 = FactoryBot.create(:valid_data_attribute_internal_attribute, attribute_subject: c, predicate: p1 )
-          d2 = FactoryBot.create(:valid_data_attribute_internal_attribute, attribute_subject: c, predicate: p3 )
-          d3 = FactoryBot.create(:valid_data_attribute_internal_attribute, attribute_subject: f, predicate: p2 ) # Not requested
+          # Create data with newlines and tabs - test that they get sanitized
+          # Insert directly into database to ensure we have the raw values
+          value_with_newline = "Line 1\nLine 2\nLine 3"
+          value_with_tab = "Column1\tColumn2\tColumn3"
+          value_with_both = "Text with\ttab and\nnewline"
 
-          a = Export::Dwca::Data.new(core_scope: scope, predicate_extensions: {collecting_event_predicate_id: predicate_ids } )
+          conn = ActiveRecord::Base.connection
+          conn.execute("INSERT INTO data_attributes (type, attribute_subject_id, attribute_subject_type, controlled_vocabulary_term_id, value, project_id, created_by_id, updated_by_id, created_at, updated_at) VALUES ('InternalAttribute', #{c.id}, 'CollectingEvent', #{p1.id}, #{conn.quote(value_with_newline)}, #{c.project_id}, #{c.created_by_id}, #{c.updated_by_id}, NOW(), NOW())")
```

Comment:
Mmm- factories or seed dbs? We don't populate vial SQL alone in production, we shouldn't do it in tests.